packages/cactus-plugin-satp-hermes/src/main/typescript/plugin-satp-hermes-gateway.ts

```diff
@@ -266,32 +263,32 @@ export class SATPGateway implements IPluginWebService, ICactusPlugin {
       pluginOptions.keyPair = Secp256k1Keys.generateKeyPairsBuffer();
     }
 
-    const id = uuidv4();
     if (!pluginOptions.gid) {
       pluginOptions.gid = {
-        id: id,
+        id: uuidv4(),
         pubKey: bufArray2HexStr(pluginOptions.keyPair.publicKey),
-        name: id,
+        name: uuidv4(),
```

Comment:
can't we give a name and not give and gid?
Shouldn't they be correlated?

Reply:
Sure, I can remove this